store/store_test.go

```diff
@@ -3111,9 +3180,14 @@ func (t *remoteRepoTestSuite) TestUbuntuStorePaymentMethodsRefreshesDeviceAuth(c
 			io.WriteString(w, `{"nonce": "1234567890:9876543210"}`)
 		case "/identity/api/v1/sessions":
 			authorization := r.Header.Get("X-Device-Authorization")
-			c.Check(authorization, Equals, expiredAuth)
-			io.WriteString(w, `{"macaroon": "refreshed-session-macaroon"}`)
-			refreshSessionRequested = true
+			if authorization == "" {
+				io.WriteString(w, `{"macaroon": "expired-session-macaroon"}`)
+				deviceSessionRequested = true
+			} else {
+				c.Check(authorization, Equals, expiredAuth)
+				io.WriteString(w, `{"macaroon": "refreshed-session-macaroon"}`)
+				refreshSessionRequested = true
```

Comment:
The testing here is completely out of control. Every small change in the store grows an exponential amount of boilerplate which is irrelevant for the test at hand. Post RTM and GA needs, it'd be nice to have a look at this, perhaps trying to clean this up via a fake store implementation that we can use from all these tests at once, tuning as needed.